extensions/amp-recaptcha-input/0.1/amp-recaptcha-input.js

```diff
@@ -97,12 +97,18 @@ export class AmpRecaptchaInput extends AMP.BaseElement {
     });
   }
 
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.NODISPLAY;
+  }
+
   /** @override */
   layoutCallback() {
     if (!this.registerPromise_ && this.sitekey_) {
       this.registerPromise_ = this.recaptchaService_.register(this.sitekey_);
     }
-    return this.registerPromise_;
+
+    return /** @type {!Promise} */ (this.registerPromise_);
```

Comment:
Why type conversion?

Reply:
I've tried playing around with the `registerPromise`, by wrapping in an if statement and everything, but it still thinks that I may return null here, even though I have the if statement above. Thus, I need to cast it to Promise.